### PR TITLE
Update create_deployment notebook

### PR DESCRIPTION
**Pull request by djkonro**

### CODE REVIEW

File: examples/notebooks/create_deployment.ipynb [hunk 1]
```diff
@@ -12,48 +23,288 @@
...
-   "execution_count": null,
+   "execution_count": 2,
    "metadata": {
     "collapsed": true
    },
    "outputs": [],
    "source": [
-    "client.Configuration().host=\"http://localhost:8080\""
+    "config.load_kube_config()\n",
+    "extensions = client.ExtensionsV1beta1Api()"
```
Comment (pokoli) [inline examples/notebooks/create_deployment.ipynb]: I also changed this line in:

https://github.com/kubernetes-incubator/client-python/pull/242/files

I think it's better to load incluster config as the examples are intended to be run from inside a kubernetes cluster.

Comment (sebgoa) [inline examples/notebooks/create_deployment.ipynb]: @pokoli do you mind if we merge this PR and close yours. This addresses more issues in the notebook. Once merged we can discuss more.

The general issue with these notebooks is to figure out a way to quickly test them.

I am tempted to add two steps in the notebooks with a try loop that loads in cluster and loads out of cluster.

Comment (sebgoa) [inline examples/notebooks/create_deployment.ipynb]: @mbohlool I am +1 on this, we need to improve things but it is a first step

Comment (pokoli) [inline examples/notebooks/create_deployment.ipynb]: Makes sense to merge my PR into this. When I first created my PR I wasn't aware of this. So please go ahead.

File: examples/notebooks/create_deployment.ipynb [hunk 2]
```diff
@@ -12,48 +23,288 @@
...
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 3,
    "metadata": {
     "collapsed": true
    },
    "outputs": [],
    "source": [
-    "extensions = client.ExtensionsV1beta1Api()"
+    "deployment = client.models.extensions_v1beta1_deployment.ExtensionsV1beta1Deployment()"
```
Comment (sebgoa) [inline examples/notebooks/create_deployment.ipynb]: @djkonro actually you should change this to `deployment = client.ExtensionsV1beta1Deployment()`

File: examples/notebooks/create_deployment.ipynb [hunk 3]
```diff
@@ -12,48 +23,288 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "#extensions.delete_namespaced_deployment(name='nginx-deployment', namespace='default', body=client.V1DeleteOptions(),grace_period_seconds = 56, propagation_policy =  \"Background\")"
```
Comment (mbohlool) [inline examples/notebooks/create_deployment.ipynb]: Search issues for this. I think somebody had the same problem recently and we discussed it there. Briefly, I think there is a bug upstream (in k8s itself) and you need to delete childs one by one.

Comment (sebgoa) [inline examples/notebooks/create_deployment.ipynb]: I don't think it is a bug. The kubectl delete deployment has a reapper that cascades deletion to pods, but the delete deployment API call itself does not do that.

@djkonro you can list pods with the label of the deployment and delete them based on that.

Comment (mbohlool) [inline examples/notebooks/create_deployment.ipynb]: ref: https://github.com/kubernetes-incubator/client-python/issues/162

Look like with 2.0.0, you should be able to do cascade delete?

Comment (djkonro) [inline examples/notebooks/create_deployment.ipynb]: @mbohlool, I have tried various options but it does not seem to delete dependents, even when I set `propagationPolicy` to `Background` or `Foreground`, like in the function call below.
`extensions.delete_namespaced_deployment(name='nginx-deployment', namespace='default', body=client.V1DeleteOptions(),grace_period_seconds = 56, propagation_policy =  \"Background\")`

Comment (djkonro) [inline examples/notebooks/create_deployment.ipynb]: Even the `grace_period_second` does not seem to take effect.

Comment (djkonro) [inline examples/notebooks/create_deployment.ipynb]: @sebgoa The pods are a part of a ReplicaSet, so they automatically get recreated when I delete them. I usually just tend to deleted the ReplicaSet in order to delete the pods, but I am looking for a better solution where I can delete them directly from the deployment.

Comment (djkonro) [inline examples/notebooks/create_deployment.ipynb]: Thanks to @vascofg, updating my version of Kubernetes solved my cascading deletion issue.

**Comment by djkonro** [comment]
@sebgoa, I have added RollingUpdate and RollBack to the deployment notebook, as seen in https://kubernetes.io/docs/concepts/workloads/controllers/deployment/. Please let me know what you think.

**Comment by sebgoa** [comment]
@djkonro there is no rolling update or rollback, I think you forgot to push the commit.

**Comment by djkonro** [comment]
@sebgoa  Thanks for noticing the omission, sorry I did not realize the commit did not contain rolling update or rollback.

**Comment by sebgoa** [comment]
/lgtm

@mbohlool

**Pull request merged**